packages/http-client-csharp/generator/Microsoft.Generator.CSharp.Input/src/InputTypes/InputEnumType.cs

```diff
@@ -7,13 +7,13 @@ namespace Microsoft.Generator.CSharp.Input
 {
     public class InputEnumType : InputType
     {
-        public InputEnumType(string name, string crossLanguageDefinitionId, string? accessibility, string? deprecated, string description, InputModelTypeUsage usage, InputPrimitiveType valueType, IReadOnlyList<InputEnumTypeValue> values, bool isExtensible)
+        public InputEnumType(string name, string crossLanguageDefinitionId, string? accessibility, string? deprecated, string? summary, string? doc, InputModelTypeUsage usage, InputPrimitiveType valueType, IReadOnlyList<InputEnumTypeValue> values, bool isExtensible)
             : base(name)
         {
             CrossLanguageDefinitionId = crossLanguageDefinitionId;
             Accessibility = accessibility;
             Deprecated = deprecated;
-            Description = description;
+            Description = string.IsNullOrEmpty(summary) ? (doc ?? string.Empty) : summary;
```

Comment:
I think we should change this property to two new properties `Summary` and `Doc`
The classes in input types should just be the C# version of those typescript interfaces.

Reply:
I see two different strategies regarding the input types between `http-client-csharp` and `autorest.csharp`:
1. In `http-client-csharp`, we declare them as `class`. So, I keep the minimal interface here.
2. In `autorest.csharp`, we declare them as `record`. So, I keep the interface like the original data schema as much as possible.

Reply:
record is a class, there is no differences.
Making this change might take quite a few efforts, this I admit.
But we should not be avoiding them - I have done multiple of those when consolidating the models part.